bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -228,7 +228,7 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                     }
                     break;
                 case CHANNEL_VOLUMEDB:
-                    setVolumeDb(((QuantityType<?>) command).floatValue(), zone, this.host);
+                    setVolumeDb(Float.parseFloat(command.toString().replace(" dB", "")), zone, this.host);
```

Comment:
Perhaps it would be better to check the type of the command rather than stripping a hardcoded unit here? I believe this would throw if you would for example:
```
openhab:send VolumeItem "1 ppm"
```

You could then support `QuantityType` and `DecimalType` explicitly.

Reply:
Makes sense, however I am not sure whether it makes sense to enable UoM for Items linked to the volume dB channel.

Reply:
Something like:
```java
if (command instanceof QuantityType commandAsQuantity) {
    float volumeDB = commandAsQuantity.toUnit(Units.DECIBEL).floatValue();
} else if (command instanceof DecimalType commandAsDecimal) {
    float volumeDB = commandAsDecimal.floatValue();
}
```

Reply:
> however I am not sure whether it makes sense to enable UoM for Items linked to the volume dB channel.

Channel is already defined to support UoM:
https://github.com/openhab/openhab-addons/blob/8a67d0ad94a2b400f09b4a1ec55d789b0807b396/bundles/org.openhab.binding.yamahamusiccast/src/main/resources/OH-INF/thing/thing-types.xml#L98-L104